src/lib/netlink/bond.rs

```
@@ -53,22 +50,16 @@ struct BondAdInfo {
 const IFLA_BOND_MODE: u16 = 1;
 const IFLA_BOND_AD_INFO: u16 = 23;
 
-fn parse_as_nested_ipv4_addr(data: &[u8]) -> Vec<Ipv4Addr> {
-    let mut i: usize = 0;
-    let mut addrs = Vec::new();
-    while i < data.len() {
-        let hdr_ptr = data.as_ptr().wrapping_offset(i.try_into().unwrap());
-        let hdr = parse_nla_header(hdr_ptr);
-        let data_ptr = data
-            .as_ptr()
-            .wrapping_offset((i + NL_ATTR_HDR_LEN).try_into().unwrap());
-        let data = unsafe {
-            slice::from_raw_parts(data_ptr, hdr.nla_len - NL_ATTR_HDR_LEN)
-        };
-        addrs.push(Ipv4Addr::new(data[0], data[1], data[2], data[3]));
-        i = i + hdr.nla_len;
+fn parse_as_nested_ipv4_addr(raw: &[u8]) -> Vec<Ipv4Addr> {
+    let mut addresses = Vec::new();
+    let nlas = NlasIterator::new(raw);
+    for nla in nlas {
+        if let Ok(nla) = nla {
+            let data = nla.value();
+            addresses.push(Ipv4Addr::new(data[0], data[1], data[2], data[3]));
+        }
```

Please use `eprintln!()` for errors instead of silently ignore them.